packages/react-components/tests/robots/robot-marker.spec.tsx

```diff
@@ -0,0 +1,63 @@
+import { act, render as render_, RenderResult } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import React from 'react';
+import { ColorContext, ColorManager, RobotMarker } from '../../lib';
+import { makeRobot } from './test-utils';
+
+describe('robot-markers', () => {
+  let colorManager: ColorManager;
+  let fakeOnClick: ReturnType<typeof jasmine.createSpy>;
+
+  beforeEach(() => {
+    colorManager = new ColorManager();
+    fakeOnClick = jasmine.createSpy();
+    colorManager.robotPrimaryColor = jasmine.createSpy();
```

Comment:
I don't think we need a mock here anymore as karma should support the apis needed, you may add a FIXME and revisit it in another PR if it doesn't work.